gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java

```diff
@@ -41,7 +41,7 @@
 @Slf4j
 public class FlowStatusGenerator {
   public static final List<String> FINISHED_STATUSES = Lists.newArrayList(ExecutionStatus.FAILED.name(),
-      ExecutionStatus.COMPLETE.name(), ExecutionStatus.CANCELLED.name());
+      ExecutionStatus.COMPLETE.name(), ExecutionStatus.CANCELLED.name(), ExecutionStatus.PENDING_RETRY.name());
```

Comment:
this is used in a number of places. is it acceptable to change the semantics for all of them?  e.g. is it OK here in `MysqlDMStateStore`:
```
  public boolean hasRunningJobs(DagManager.DagId dagId) throws IOException {
    return getDagNodes(dagId).stream()
        .anyMatch(node -> !FlowStatusGenerator.FINISHED_STATUSES.contains(node.getValue().getExecutionStatus().name()));
  }
```
?

Reply:
The code is like 
```
if (jobStatus.isShouldRetry()) {
      XXX
    } else if (!dagManagementStateStore.hasRunningJobs(getDagId())) {
      YYY
    }
```
so both code parts do not collide. Also it is kinda ok for `hasRunningJobs` to return `true` when status is PENDING_RETRY, isn't it?

Reply:
hmm, actually it should not go to that warn message, because `flowEvent` field is never set to PENDING_RETRY. in fact PENDING_RETRY does not correspond to any "flow event", this is a case when GTE name does not have any execution status

Reply:
Why use this PENDING_RETRY status at all if Reevaluate is going to use the # of retry attempts to determine if anything to retry for the job? can you summarize how this particular status is used in the PR desc?

Reply:
Yea, we can actually make it work without PENDING_RETRY. That was the approach in the 1st commit (sorry, it lost because I mistakenly force pushed another squashed commit)
We can completely remove PENDING_RETRY. But in that approach,  job's status will be shown 'FAILED' between both the attempts,, Kip preferred 'PENDING_RETRY'...